shard.go

```diff
@@ -1,33 +1,48 @@
 package influxdb
 
 import (
-	"encoding/json"
+	"encoding/binary"
 	"errors"
 	"fmt"
+	"math"
+	"sort"
 	"time"
-	"unsafe"
 
 	"github.com/boltdb/bolt"
 )
 
-// Shard represents the physical storage for a given time range.
-type Shard struct {
+// ShardGroup represents a group of shards created for a single time range.
+type ShardGroup struct {
 	ID        uint64    `json:"id,omitempty"`
 	StartTime time.Time `json:"startTime,omitempty"`
 	EndTime   time.Time `json:"endTime,omitempty"`
+	Shards    []*Shard  `json:"shards,omitempty"`
+}
 
-	replicaN    []uint64 // replication factor
-	dataNodeIDs []uint64 // owner nodes
+// close closes all shards.
+func (g *ShardGroup) close() {
+	for _, sh := range g.Shards {
+		_ = sh.close()
+	}
+}
+
+// Shard represents the physical storage for a given time range.
```

Comment:
This type definition is a bit incongruent -- or perhaps I don't understand it. It's got 1 pointer to a BoltDB, but it says it lives on multiple nodes, which is obviously not possible. Strictly speaking this doesn't make sense, and is not quite in line with our definition of shard.

Or am I missing something?